File: ejercicios hechos/Python/ejercicios repeticiones/act8.py
#Programa: act8.py
#Propósito: Hacer un programa que muestre un cronometro, indicando las horas, minutos y segundos. Para hacer una espera en Python podemos usar el método sleep del módulo time.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 29/10/2019
#
#Variables a usar:
# hora,minuto y segundos
# 
#Algoritmo:
# Para las horas de 0 hasta 24
# Para los minutos desde 0 hasta 60
# Para los segundos desde 0 hasta 60
# imprimimos el resultado
# esperamos mediante time.sleep

print("Temporizador en python")
print("------------------------\n")

import time

print("Comienzo")

#Generamos un bucle y con time.sleep le pedimos que espere 1 segundo para devolver los datos
for h in range(0 , 24):
    for m in range(0 ,60):
        for s in range(0 ,60):             
            print("Hora: ", h, "Minuto: ", m, "Segundos: ", s)
            time.sleep(1)

File: ejercicios terminados/Java/estructuras repetitivas/act8.java
/*Programa: act8.java
 * 
 *Propósito: Hacer un programa que muestre un cronometro, indicando las 
 * horas, minutos y segundos.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 * 
 *
 *Diseño:
*/
import java.util.concurrent.TimeUnit;
public class act8 {
    public static void main(String args[]) throws InterruptedException {
        System.out.println("Programa que muestra un conmetro");
        System.out.println("--------------------------------");
        int hora;
        int minuto;
        int segundo;
        String borraAnterior = "\b\b\b\b\b\b\b\b"; 

        for (hora=0; hora<=23; hora++) {
          for (minuto=0; minuto<=59; minuto++) {
            for (segundo=0; segundo<=59; segundo++) {
              System.out.print(borraAnterior+hora+":"+minuto+":"+segundo);
              TimeUnit.SECONDS.sleep(1);
            }
          }
        }
    }
}
